stdlib/char.gr

```diff
@@ -153,45 +186,62 @@ let rec fromCode = (code: Number) => {
   result
 }
 
-// Returns the next valid Unicode character by code point. Fails if the input character is U+10FFFF.
-// @param char: Char - the input character
-// @returns Char
-let succ = (c) => {
-  let codePoint = code(c)
+/**
+ * Returns the next valid Unicode character by Unicode scalar value.
+ * Throws if the input character is the max valid Unicode scalar value.
+ * 
+ * @param char: The character
+ * @returns The next valid Unicode character by Unicode scalar value
+ * 
+ * @since 0.3.0
+ */
+export let succ = (char) => {
+  let codePoint = code(char)
   if (codePoint == max) {
-    fail "no valid Unicode code point past U+10FFF"
+    fail "no valid Unicode scalar value past U+10FFF"
```

Comment:
Can't catch this throw anyway, so we probably don't need to worry about the message changing. Thoughts @ospencer ?

Reply:
Right, it's fine if this changes.